java-client/src/main/java/co/elastic/clients/transport/rest_client/RestClientHttpClient.java

```diff
@@ -112,6 +119,9 @@ public CompletableFuture<Response> performRequestAsync(
         future.cancellable = restClient.performRequestAsync(restRequest, new ResponseListener() {
             @Override
             public void onSuccess(org.elasticsearch.client.Response response) {
+                if (options.keepResponseBodyOnException()) {
```

Comment:
This should be moved to `ElasticsearchTransportBase`, which will allow it to be independent of the actual `HttpClient` implementation, and also `RepeatableBodyResponse` which will then implement `TransportHttpClient.Response`.